src/events/backup.py

```diff
@@ -41,7 +47,11 @@ def __init__(self, charm):
 
         self.framework.observe(self.charm.on.create_backup_action, self._on_create_backup_action)
         self.framework.observe(self.charm.on.list_backups_action, self._on_list_backups_action)
-        # self.framework.observe(self.charm.on.restore_action, self._on_restore_action)
+        self.framework.observe(self.charm.on.restore_action, self._on_restore_action)
+
+        self.framework.observe(
+            getattr(self.charm.on, "cluster_relation_changed"), self._restore_event_dispatch
```

Comment:
**todo**: Use `PEER`, after full green CI

Reply:
Actually, we do not do that in `charm.py` either

Reply:
**question** is this overloading the relation changed with an additional call? Does this method run on top of the one defined in `charm.py`. If so, I honestly don't mind: in one hand I believe it is all right if the two methods are orthogonal one another, and don't conflict, but overloading can make debugging a bit more complex. 

Anyhow, it is probably just worth discussing

Reply:
You can bind multiple methods to an event, the ops framework simply calls them in the order they were bound. Since `BackupEvent` is instantiated before the `observe` statement in `charm.py`, the bound method from this file is the first one to run